pkg/stream/producer.go

```diff
@@ -267,133 +244,80 @@ func (producer *Producer) getStatus() int {
 func (producer *Producer) startUnconfirmedMessagesTimeOutTask() {
 
 	go func() {
-		for {
+		isActive := true
```

Comment:
I think this `isActive` is not needed, just a for without condition and a return where needed should be enough.